pkg/bindings/images/build.go

```diff
@@ -93,6 +94,19 @@ func Build(ctx context.Context, containerFiles []string, options entities.BuildO
 		params.Set("labels", l)
 	}
 
+	var (
+		headers map[string]string
+		err     error
+	)
+	if options.SystemContext.DockerAuthConfig != nil {
```

Comment:
This could be a nil deref on `options.SystemContext` so we need an additional check for that.